### ISSUE TITLE
[Engine/ScriptNode/Faust] Compiling a Faust node with vbargraph outputs doesn't preserve them in the compiled node

### ISSUE THREAD

**Issue by aaronventure** [christophhart/HISE#619]
If you use vbargraph in Faust, it will create a cable output in the node, but the cable outputs will not be present on the compiled node. 

```
something = 0.3;
somethingElse = 0.5;
intermediary = something * somethingElse : vbargraph("[0]CableOut", -1, 1);

process _,_ : par(i, 2, *(intermediary));
```

macOS 14.5
376a19962e40899f0f292e7fc9a49fdfb45670ed

**Comment by sletz** [comment]
@aaronventure: look at the `attach` primititive [here](https://faustdoc.grame.fr/manual/syntax/#attach-primitive).

**Comment by aaronventure** [comment]
Of course, it needs to be in the process chain otherwise it's not visible at all in the HISE Faust node, and attach is useful for cases where you're not already using a variable in the chain. 

But this issue pertains to HISE's handling of bargraph in compiled nodes. 

Here's a simple example of how vbargraph looks in action in HISE when used in the Faust node. The Faust node will not show parameters or bargraph outputs if they're not in the process chain, so there's no error there. 
<img width="286" alt="image" src="https://github.com/user-attachments/assets/1c9b769d-bfc2-4f65-9ec6-a5f10427b008" />

But in order to actually use that Faust program in the final product, one first needs to compile the node, and if you then want to use it in ScriptNode with other nodes, you need to add the compiled node from the Project tab in ScriptNode (which means that the compilation of the Faust nodes should take place in a separate network which is used specifically for that). Once you have the compiled node, it looks like this. There's no vbargraph cable output.

<img width="295" alt="image" src="https://github.com/user-attachments/assets/8ee4cc63-7c50-497a-b0b5-30ea8b5a423b" />

The code was
```
import("stdfaust.lib");

gain = hslider("Gain", 1, 0, 1, 0.001) : si.smoo;

diff = 1 - gain : vbargraph("Difference", 0, 1);

process = _, _ : si.bpar(2, attach(*(gain), diff));
```

**Comment by trummerschlunk** [comment]
This would be very helpful and appreciated, as it will make HISE a great shell for Faust programs.
+1 <3

**Comment by sletz** [comment]
I'm still not clear what has to be done as I understand this is a HISE integration issue yes ?

**Comment by aaronventure** [comment]
> I'm still not clear what has to be done as I understand this is a HISE integration issue yes ?

Probably, yeah, because bargraph works fine until the node gets compiled in HISE and converted to a standalone ScriptNode node. Christoph will confirm.

**Comment by christoph-hart** [comment]
That's not a Faust issue, but precisely what the public_mod node is for:

https://docs.hise.audio/scriptnode/list/routing/public_mod.html

So connect the faust node cable output to this node, compile it and it should show up correctly.

![Image](https://github.com/user-attachments/assets/e9ae4fd9-ca18-48b4-b1c0-80f60f9a2356)

Faust code to use (modified your example to take a parameter input)

```faust
// Faust Source File: public_test
// Created with HISE on 2025-01-20
import("stdfaust.lib");



gain = hslider("gain", 0.0, 0.0, 1.0, 0.01);

intermediary = 1.0 * gain : vbargraph("[0]CableOut", -1, 1);

process = _,_ : par(i, 2, *(intermediary));
```

**Comment by aaronventure** [comment]
Alright, that works. This opens up another dimension in the Faust compilation UX issue I talked about on the forum, and I'll expand on it there. I'll close this issue, thanks for responding.
